### PR TITLE
feat: add adjust_bound_pos method to Plan

### PR DESCRIPTION
**Pull request by snowyu**

### CODE REVIEW

File: plan.lua [hunk 1]
```diff
@@ -104,22 +129,9 @@ end
...
-		self.data.max_pos.x = plan_pos.x
-	end
-	if not self.data.min_pos.x or plan_pos.x < self.data.min_pos.x then
-		self.data.min_pos.x = plan_pos.x
-	end
-	if not self.data.max_pos.z or plan_pos.z > self.data.max_pos.z then
-		self.data.max_pos.z = plan_pos.z
-	end
-	if not self.data.min_pos.z or plan_pos.z < self.data.min_pos.z then
+  self:adjust_bound_pos(plan_pos)
```
Comment (bell07) [inline plan.lua:132]: The code is formatted using tabs. I see you use spaces. I know the spaces are recommended, but lot of Mintest mods uses tabs. Please adjust the spacing to tabs

File: plan.lua [hunk 2]
```diff
@@ -104,22 +129,9 @@ end
...
+
+  if not self.data.min_pos.z or plan_pos.z < self.data.min_pos.z then
```
Comment (bell07) [inline plan.lua:134]: This lines are moved to adjust_bound_pos. Is there a reason to keep the lines redundant in adjust_building_info?

File: plan.lua [hunk 3]
```diff
@@ -46,6 +46,31 @@ function plan.new(plan_id , anchor_pos)
 	return self -- the plan object
 end
 
+--------------------------------------
+-- Adjust bound min/max position information
+--------------------------------------
+function plan_class:adjust_bound_pos(plan_pos)
```
Comment (bell07) [inline plan.lua:52]: Please add the function to API documentation in README.md

File: plan.lua (hunk 1 above)
Comment (snowyu) [reply plan.lua:132]: ok

File: plan.lua (hunk 2 above)
Comment (snowyu) [reply plan.lua:134]: Sorry, my mistake.

**Comment by bell07** [comment]
Thank you!

**Pull request merged**